pkg/secrets/pushtofile/retrieve_secrets.go

```diff
@@ -0,0 +1,62 @@
+package pushtofile
+
+import (
+	"strings"
+
+	"github.com/cyberark/conjur-authn-k8s-client/pkg/access_token"
+	"github.com/cyberark/conjur-authn-k8s-client/pkg/log"
+	"github.com/cyberark/secrets-provider-for-k8s/pkg/log/messages"
+	"github.com/cyberark/secrets-provider-for-k8s/pkg/secrets/clients/conjur"
+)
+
+//var retrieveConjurSecretsFunc = conjur.RetrieveConjurSecrets
+
+// IterateOverGroups parses the SecretsGroup, gets
+// the secrets from Conjur and updates the SecretsGroup with the secret
+func IterateOverGroups(accessToken access_token.AccessToken,
+	secretGroups *SecretGroups) error {
+
+	return iterateOverGroups(accessToken, secretGroups,
+		conjur.RetrieveConjurSecrets)
+
+}
+
+func iterateOverGroups(accessToken access_token.AccessToken,
+	secretGroups *SecretGroups,
+	retrieveConjurSecretsFunc conjur.RetrieveConjurSecretsFunc) error {
+
+	var err error
+	var retrieved map[string][]byte
+
+	for groupkey, group := range *secretGroups {
```

Comment:
Some thoughts on
>`retrieveConjurSecretGroup` is being called for each group separately and in sequence.

1. We could make this code concurrent by using `sync.Waitgroup`, see https://gobyexample.com/waitgroups, and running each fetch in a separate Go routine. For capturing results and errors you can use a channel, see this for errors https://stackoverflow.com/a/24715755/5458056.
2. We could also leverage the fact that secrets across all groups can be fetched in one fell swoop, and only have a single call to the `retrieveConjurSecretGroup` function.